project-2-Set-Game/SetGame.swift

```diff
@@ -0,0 +1,278 @@
+//
+//  SetGame.swift
+//  project-2-Set-Game
+//
+//  Created by Tomer Kobrinsky on 13/02/2019.
+//  Copyright © 2019 Tomer Kobrinsky. All rights reserved.
+//
+
+import Foundation
+
+class SetGame {
+    private(set) var selectedCards = [Card]()
+    private(set) var deck = [Card]()
+    private(set) var cardsOnBoard = [Card]()
+    private(set) var selectedCardsMatched: Bool?
+    private(set) var score = 0
+    private var timeWhenGameStarted = Date.init()
+    private(set) var opponentState = OpponentState.notWaitingForTurn
+    private(set) var opponentScore = 0
+    private(set) var isGameOver = false
+    private var opponentCycleTimer,opponentWaitTimer, opponentReadyToMakeAMoveTimer , opponentMakesAMoveTimer, opponentDoesntWaitForTurnTimer : Timer?
+    
+    private var timeCycle = 0
+    weak var delegate: SetGameDelegate?
+    init() {
+        createDeck()
+        deck.shuffle()
+        dealTwelveCards()
+        setOppnentTimer()
+    }
+    func resetGame() {
+        isGameOver = false
+        selectedCards =  [Card]()
+        cardsOnBoard = [Card]()
+        deck = [Card]()
+        createDeck()
```

Comment:
the logic in lines 36-38 and in line 46 is identical to lines 26-29
maybe use a function?